### PR TITLE
[Bugfix] Fix numerical stability of `lie_closure`

### PR DESCRIPTION
**Pull request by Qottmann**
Noticed that `lie_closure` was not numerically stable after a [user report](https://discuss.pennylane.ai/t/calculation-of-dynamical-lie-algebra/4309/3)

Fixing that by

- [x] using the slightly slower but more stable `qml.math.linalg.solve` over `qml.math.linalg.inv`
- [x] exposing `tol` as kwarg to the user
- [x] setting the default of `tol` higher (two orders of magnitude higher as the machine precision of `_M`'s dtype)

### CODE REVIEW

File: pennylane/pauli/dla/lie_closure.py [hunk 1]
```diff
@@ -47,6 +48,7 @@ def lie_closure(
         pauli (bool): Indicates whether it is assumed that :class:`~.PauliSentence` or :class:`~.PauliWord` instances are input and returned.
             This can help with performance to avoid unnecessary conversions to :class:`~pennylane.operation.Operator`
             and vice versa. Default is ``False``.
+        tol (float): Numerical tolerance for linear independence check; :math:`v - M (M^T M)^{-1} M^T v > \text{tol}`.
```
Comment (trbromley) [inline pennylane/pauli/dla/lie_closure.py:51]:
```suggestion
        tol (float): Numerical tolerance for the linear independence check used in :class:`~.PauliVSpace`.
```
Maybe better to mention there as the `lie_closure` page has less context on $v$ and $M$.

File: pennylane/pauli/dla/lie_closure.py [hunk 2]
```diff
@@ -195,6 +197,7 @@ class PauliVSpace:
     Args:
         generators (Iterable[Union[PauliWord, PauliSentence, Operator]]): Operators that span the vector space.
         dtype (type): ``dtype`` of the underlying DOK sparse matrix ``M``. Default is ``float``.
+        tol (float): Numerical tolerance for the linear independence check; :math:`v - M (M^T M)^{-1} M^T v > \text{tol}`.
```
Comment (trbromley) [inline pennylane/pauli/dla/lie_closure.py:200]:
```suggestion
        tol (float): Numerical tolerance for the linear independence check. If
        :math:`v - M (M^T M)^{-1} M^T v > \text{tol}` for a vector :math:`v` then
        it is deemed to be linearly independent from the set of vectors composing
        the rows of the matrix :math:`M`.
```
This explains it to me, but could be wrong?

**Review by trbromley — APPROVED** [review]
Thanks @Qottmann! Approved, subject to my comments

File: pennylane/pauli/dla/lie_closure.py (hunk 2 above)
Comment (vincentmr) [reply pennylane/pauli/dla/lie_closure.py:200]: We could also say: "If the norm of the projection of `v` onto `M^\perp` is greater than `tol`, then
	        it is deemed to be linearly independent."

**Review by vincentmr — APPROVED** [review]
Just a suggestion, but otherwise looks good to me. Cheers @Qottmann .

File: pennylane/pauli/dla/lie_closure.py (hunk 2 above)
Comment (Qottmann) [reply pennylane/pauli/dla/lie_closure.py:200]: I like that ✅

**Pull request merged**